runtime/near-vm-runner/src/logic/logic.rs

```diff
@@ -30,14 +30,178 @@ fn base64(s: &[u8]) -> String {
     base64::engine::general_purpose::STANDARD.encode(s)
 }
 
+/// Structure representing the results and outcomes of a contract execution.
+///
+/// This is a subset of [`VMLogic`] that's strictly necessary to produce `VMOutcome`s.
+pub struct ExecutionResultState {
+    /// All gas and economic parameters required during contract execution.
+    config: Arc<Config>,
```

Comment:
Hmm I’m not sure I get why gas parameters would be required to build a `VMOutcome`. AFAIU, this is only required in order to actually execute the contract, and not really for the result.

So WDYT about naming this struct `ExecutionState` rather than `ExecutionResultState`?

Reply:
The config (or rather `ext_costs` within it) is necessary for the profile data computation. Since `ext_costs` always gets constructed as part of an overall `Config`, I didn't see a good reason to split it up anyhow. In principle `ExecutionResultState::compute_outcome` was the guiding factor on what to include here and what to leave back at `VMLogic`.

Reply:
(The comment also explains why this is specifically a "ResultState" -- it contains things that are strictly only for the eventual computation of a `VMOutcome`)